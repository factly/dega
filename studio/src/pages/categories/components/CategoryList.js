import React from 'react';
import { Popconfirm, Button, Table, Space } from 'antd';

import { useDispatch } from 'react-redux';
import { deleteCategory } from '../../../actions/categories';
import { Link } from 'react-router-dom';
import { DeleteOutlined } from '@ant-design/icons';

function CategoryList({ actions, data, filters, setFilters, fetchCategories }) {
  const dispatch = useDispatch();
  const columns = [
    {
      title: 'Name',
      dataIndex: 'name',
      key: 'name',
      render: (_, record) => {
        return (
          <Link
            style={{
              marginRight: 8,
            }}
            to={`/categories/${record.id}/edit`}
          >
            {record.name}
          </Link>
        );
      },
    },
    { title: 'Slug', dataIndex: 'slug', key: 'slug' },
    { title: 'Parent Category', dataIndex: ['parent_category', 'name'], key: 'parent_id' },
    {
      title: 'Action',
      dataIndex: 'operation',
      fixed: 'right',
      align: 'center',
      width: 150,
      render: (_, record) => {
        return (
          <Popconfirm
            title="Are you sure you want to delete this?"
            onConfirm={() => dispatch(deleteCategory(record.id)).then(() => fetchCategories())}
            disabled={!(actions.includes('admin') || actions.includes('delete'))}
          >
            <Button
              icon={<DeleteOutlined />}
              disabled={!(actions.includes('admin') || actions.includes('delete'))}
              type="danger"
            />
          </Popconfirm>
        );
      },
    },
  ];

  return (
    <Space direction={'vertical'}>
      <Table
        bordered
        columns={columns}
        dataSource={data.categories}
        loading={data.loading}
        rowKey={'id'}
        pagination={{
          total: data.total,
          current: filters.page,
          pageSize: filters.limit,
          onChange: (pageNumber, pageSize) =>
            setFilters({ ...filters, page: pageNumber, limit: pageSize }),
          pageSizeOptions: ['10', '15', '20'],
        }}
      />
    </Space>
  );
}

export default CategoryList;
