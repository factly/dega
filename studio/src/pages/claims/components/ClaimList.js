import React from 'react';
import { Popconfirm, Button, Table, Space } from 'antd';
import { useDispatch } from 'react-redux';
import { deleteClaim } from '../../../actions/claims';
import { Link } from 'react-router-dom';
 import dayjs from 'dayjs'
import { DeleteOutlined } from '@ant-design/icons';

function ClaimList({ actions, data, filters, fetchClaims, onPagination }) {
  const dispatch = useDispatch();
  const columns = [
    {
      title: 'Claim',
      dataIndex: 'claim',
      key: 'claim',
      width: '30%',
      render: (_, record) => {
        return (
          <Link
            style={{
              marginRight: 8,
            }}
            to={`/claims/${record.id}/edit`}
          >
            {record.claim}
          </Link>
        );
      },
    },
    { title: 'Claimant', dataIndex: 'claimant', key: 'claimant', width: '20%' },
    { title: 'Rating', dataIndex: 'rating', key: 'rating', width: '20%' },
    {
      title: 'Claim Date',
      dataIndex: 'claim_date',
      width: '20%',
      render: (_, record) => {
        return (
          <span title={record.claim_date}>
            {record.claim_date ? dayjs(record.claim_date).format('MMMM Do YYYY') : null}
          </span>
        );
      },
    },
    {
      title: 'Action',
      dataIndex: 'operation',
      fixed: 'right',
      align: 'center',
      width: 150,
      render: (_, record) => {
        return (
          <Popconfirm
            title="Are you sure you want to delete this?"
            onConfirm={() => dispatch(deleteClaim(record.id)).then(() => fetchClaims())}
            disabled={!(actions.includes('admin') || actions.includes('delete'))}
          >
            <Button
              icon={<DeleteOutlined />}
              disabled={!(actions.includes('admin') || actions.includes('delete'))}
              danger
            />
          </Popconfirm>
        );
      },
    },
  ];

  return (
    <Space direction="vertical">
      <Table
        bordered
        columns={columns}
        dataSource={data.claims}
        loading={data.loading}
        rowKey={'id'}
        pagination={{
          showTotal: (total, range) => `${range[0]}-${range[1]} of ${total} results`,
          total: data.total,
          current: filters.page,
          pageSize: filters.limit,
          onChange: (pageNumber, pageSize) => onPagination(pageNumber, pageSize),
          pageSizeOptions: ['10', '15', '20'],
        }}
      />
    </Space>
  );
}

export default ClaimList;
