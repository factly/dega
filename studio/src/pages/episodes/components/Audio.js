import React from 'react';
import UppyUploader from '../../../components/Uppy';
import { Modal, Space, Button } from 'antd';
import { UploadOutlined } from '@ant-design/icons';
const Audio = ({ url, onUpload }) => {
  const [show, setShow] = React.useState(false);
  return (
    <>
      <Modal
        open={show}
        onCancel={() => setShow(false)}
        closable={false}
        width={'800px'}
        footer={[
          <Button key="back" onClick={() => setShow(false)}>
            Cancel
          </Button>,
          <Button
            key="submit"
            type="primary"
            onClick={() => {
              setShow(false);
            }}
          >
            Submit
          </Button>,
        ]}
      >
        <Space direction="vertical">
          <UppyUploader
            onUpload={(values) => {
              onUpload(values[0]?.url?.raw);
              setShow(false);
            }}
            allowedFileTypes={['audio/*']}
          />
        </Space>
      </Modal>
      <Space direction="vertical">
        {url ? (
          <audio controls>
            <source src={url} type="audio/ogg" />
          </audio>
        ) : null}
        <Space direction="horizontal">
          {url ? <Button onClick={() => onUpload('')}>Remove</Button> : null}{' '}
          <Button
            icon={<UploadOutlined />}
            onClick={() => setShow(true)}
          >Upload</Button>
        </Space>
      </Space>
    </>
  );
};

export default Audio;
