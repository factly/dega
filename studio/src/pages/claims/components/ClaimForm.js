import React from 'react';
import { Button, Form, Input, DatePicker, Row, Col, Collapse } from 'antd';
import Selector from '../../../components/Selector';
import { maker } from '../../../utils/sluger';
 import dayjs from 'dayjs'
import { MinusCircleOutlined, PlusOutlined } from '@ant-design/icons';
import getJsonValue from '../../../utils/getJsonValue';
import { DescriptionInput, MetaForm, SlugInput } from '../../../components/FormItems';

const layout = {
  // labelCol: {
  //   span: 8,
  //   offset: 2,
  // },
  // wrapperCol: {
  //   span: 20,
  //   offset: 2,
  // },
};

const ClaimForm = ({ onCreate, data = {} }) => {
  if (data && data.meta_fields) {
    if (typeof data.meta_fields !== 'string') {
      data.meta_fields = JSON.stringify(data.meta_fields);
    }
  }
  const [form] = Form.useForm();
  const [valueChange, setValueChange] = React.useState(false);
  const { Panel } = Collapse;
  const [activeKey, setActiveKey] = React.useState(['1', '2']);

  const onReset = () => {
    form.resetFields();
  };

  const disabledDate = (current) => {
    return current.valueOf() > Date.now();
  };

  const onSave = (values) => {
    values.claimant_id = values.claimant || 0;
    values.rating_id = values.rating || 0;
    values.claim_date = values.claim_date
      ? dayjs(values.claim_date).format('YYYY-MM-DDTHH:mm:ssZ')
      : null;
    values.checked_date = values.checked_date
      ? dayjs(values.checked_date).format('YYYY-MM-DDTHH:mm:ssZ')
      : null;

    onCreate(values);
  };

  const onClaimChange = (string) => {
    if (string.length > 150) {
      form.setFieldsValue({
        slug: maker(string.substring(0, 150)),
      });
    } else {
      form.setFieldsValue({
        slug: maker(string),
      });
    }
  };

  if (data && data.id) {
    data.claim_date = data.claim_date ? dayjs(data.claim_date) : null;
    data.checked_date = data.checked_date ? dayjs(data.checked_date) : null;
  }

  const handleCollapse = (props) => {
    setActiveKey(props);
  };

  return (
    <div>
      <Form
        {...layout}
        form={form}
        initialValues={data}
        name="create-claim"
        onFinish={(values) => {
          if (values.meta_fields) {
            values.meta_fields = getJsonValue(values.meta_fields);
          }
          onSave(values);
          onReset();
        }}
        onFinishFailed={(errors) => {
          let name = errors.errorFields[0].name[0];
          if (['claim', 'slug', 'claimant', 'rating'].includes(name)) {
            setActiveKey(['1']);
          } else {
            setActiveKey(['2']);
          }
          if (errors.errorFields[0].name[0] !== 'review_sources') {
          }
          if (errors.errorFields[0].name[0] !== 'claim_sources') {
          }
        }}
        onValuesChange={() => {
          setValueChange(true);
        }}
        scrollToFirstError={true}
        style={{
          paddingTop: '24px',
        }}
        layout="vertical"
      >
        <Form.Item>
          <Button
            disabled={!valueChange}
            type="primary"
            htmlType="submit"
            style={{ marginLeft: 'auto', display: 'block' }}
          >
            {data && data.id ? 'Update' : 'Submit'}
          </Button>
        </Form.Item>
        <Collapse
          style={{ width: '100%', marginBottom: '15px', maxWidth: 800, margin: '0 auto' }}
          defaultActiveKey={['1', '2']}
          activeKey={activeKey && activeKey}
          onChange={(props) => handleCollapse(props)}
          expandIconPosition="right"
          expandIcon={({ isActive }) => <Button>{isActive ? 'Close' : 'Expand'}</Button>}
        >
          <Panel header="Basic" key="1">
            <Form.Item
              name="claim"
              label="Claim"
              rules={[
                {
                  required: true,
                  message: 'Please input the Claim!',
                },
                { max: 5000, message: 'Claim must be maximum 5000 characters.' },
              ]}
            >
              <Input.TextArea
                rows={6}
                placeholder="Enter claim...."
                onChange={(e) => onClaimChange(e.target.value)}
              />
            </Form.Item>
            <SlugInput />
            <Form.Item name="fact" label="Fact">
              <Input.TextArea rows={6} placeholder={'Enter Fact ...'} />
            </Form.Item>
            <Form.Item
              name="claimant"
              label="Claimant"
              rules={[
                {
                  required: true,
                  message: 'Please add claimant!',
                },
              ]}
            >
              <Selector action="Claimants" />
            </Form.Item>

            <Form.Item
              name="rating"
              label="Rating"
              rules={[
                {
                  required: true,
                  message: 'Please add rating!',
                },
              ]}
            >
              <Selector action="Ratings" />
            </Form.Item>
            <Form.Item>
              <Form.Item
                name="claim_date"
                label="Claim Date"
                style={{ display: 'inline-block', width: 'calc(50% - 12px)' }}
              >
                <DatePicker disabledDate={disabledDate} />
              </Form.Item>
              <Form.Item
                name="checked_date"
                label="Checked Date"
                style={{ display: 'inline-block', width: 'calc(50% - 12px)' }}
              >
                <DatePicker disabledDate={disabledDate} />
              </Form.Item>
            </Form.Item>
            <DescriptionInput
              inputProps={{ placeholder: 'Enter Description...' }}
              initialValue={data.description?.json}
            />
          </Panel>
          <Panel header="Sources" key="2">
            <Form.Item label="Claim Sources">
              <Form.List name="claim_sources" label="Claim sources">
                {(fields, { add, remove }) => (
                  <>
                    {fields.map((field) => (
                      <Row
                        style={{ justifyContent: 'center', alignItems: 'baseline' }}
                        key={field}
                        gutter={13}
                      >
                        <Col span={11}>
                          <Form.Item
                            {...field}
                            name={[field.name, 'url']}
                            fieldKey={[field.fieldKey, 'url']}
                            rules={[{ required: true, message: 'Url required' }]}
                            wrapperCol={24}
                          >
                            <Input placeholder="Enter url" />
                          </Form.Item>
                        </Col>
                        <Col span={12}>
                          <Form.Item
                            {...field}
                            name={[field.name, 'description']}
                            fieldKey={[field.fieldKey, 'description']}
                            rules={[{ required: true, message: 'Description required' }]}
                            wrapperCol={24}
                          >
                            <Input placeholder="Enter description" />
                          </Form.Item>
                        </Col>
                        <MinusCircleOutlined onClick={() => remove(field.name)} />
                      </Row>
                    ))}
                    <Form.Item>
                      <Button type="dashed" onClick={() => add()} block icon={<PlusOutlined />}>
                        Add Claim sources
                      </Button>
                    </Form.Item>
                  </>
                )}
              </Form.List>
            </Form.Item>
            <Form.Item label="Review Sources">
              <Form.List name="review_sources" label="Review sources">
                {(fields, { add, remove }) => (
                  <>
                    {fields.map((field) => (
                      <Row
                        style={{ justifyContent: 'center', alignItems: 'baseline' }}
                        gutter={13}
                        key={field}
                      >
                        <Col span={11}>
                          <Form.Item
                            {...field}
                            name={[field.name, 'url']}
                            fieldKey={[field.fieldKey, 'url']}
                            rules={[{ required: true, message: 'Url required' }]}
                            wrapperCol={24}
                          >
                            <Input placeholder="Enter url" />
                          </Form.Item>
                        </Col>
                        <Col span={12}>
                          <Form.Item
                            {...field}
                            name={[field.name, 'description']}
                            fieldKey={[field.fieldKey, 'description']}
                            rules={[{ required: true, message: 'Description required' }]}
                            wrapperCol={24}
                          >
                            <Input placeholder="Enter description" />
                          </Form.Item>
                        </Col>
                        <MinusCircleOutlined onClick={() => remove(field.name)} />
                      </Row>
                    ))}
                    <Form.Item>
                      <Button type="dashed" onClick={() => add()} block icon={<PlusOutlined />}>
                        Add Review sources
                      </Button>
                    </Form.Item>
                  </>
                )}
              </Form.List>
            </Form.Item>
          </Panel>
        </Collapse>
        <MetaForm
          style={{ width: '100%', marginBottom: '15px', maxWidth: 800, margin: '0 auto' }}
        />
      </Form>
    </div>
  );
};

export default ClaimForm;
