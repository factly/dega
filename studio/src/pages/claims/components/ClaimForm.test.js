import React from 'react';
import { Provider, useSelector, useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { act } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import { DatePicker, Input, Collapse } from 'antd';
import moment from 'moment';
import { MinusCircleOutlined } from '@ant-design/icons';

import '../../../matchMedia.mock';
import ClaimCreateForm from './ClaimForm';
import { SlugInput } from '../../../components/FormItems';

jest.mock('@editorjs/editorjs');
jest.mock('react-monaco-editor', () => {
  const MonacoEditor = () => <div />;
  return MonacoEditor;
});
jest.mock('../../../actions/claimants', () => ({
  ...jest.requireActual('../../../actions/claimants'),
  addClaimant: jest.fn(),
}));

const data = {
  claim: 'title',
  slug: 'slug',
  claimant: 1,
  rating: 1,
  claim_date: moment(new Date('2020-12-12')),
  checked_date: moment(new Date('2020-12-12')),
  claim_sources: 'claim_sources',
  fact: 'review',
  review_tag_line: 'review_tag_line',
  review_sources: 'review_sources',
  description: {
    time: 1595747741807,
    blocks: [
      {
        type: 'header',
        data: {
          text: 'Editor.js',
          level: 2,
        },
      },
      {
        type: 'paragraph',
        data: {
          text:
            'Hey. Meet the new Editor. On this page you can see it in action — try to edit this text.',
        },
      },
    ],
    version: '2.18.0',
  },
};

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let onCreate, store;

describe('Claims Create Form component', () => {
  store = mockStore({
    claims: {
      req: [],
      details: {},
      loading: true,
    },
    claimants: {
      req: [],
      details: {},
      loading: true,
    },
    rating: {
      req: [],
      details: {},
      loading: true,
    },
    media: {
      req: [],
      details: {},
      loading: true,
    },
  });
  useDispatch.mockReturnValue(jest.fn(() => Promise.resolve({})));
  useSelector.mockImplementation((state) => ({ details: [], total: 0, loading: false }));

  describe('snapshot testing', () => {
    beforeEach(() => {
      onCreate = jest.fn();
      onCreate.mockImplementationOnce(
        (values) => new Promise((resolve, reject) => resolve(values)),
      );
    });
    it('should render the component', () => {
      const tree = shallow(
        <Provider store={store}>
          <ClaimCreateForm />
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
    it('should match component with data', () => {
      const tree = shallow(
        <Provider store={store}>
          <ClaimCreateForm onCreate={onCreate} data={data} />
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper, props;
    beforeEach(() => {
      props = {
        onCreate: jest.fn(),
        data: data,
      };
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <ClaimCreateForm {...props} />
          </Provider>,
        );
      });
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should not submit form with empty data', (done) => {
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <ClaimCreateForm onCreate={props.onCreate} />
          </Provider>,
        );
      });

      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.props().disabled).toBe(true);
      });
      wrapper.update();

      setTimeout(() => {
        expect(props.onCreate).not.toHaveBeenCalled();
        done();
      }, 0);
    });
    it('should submit form with given data', (done) => {
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Submit');
        submitButtom.simulate('submit');
      });
      wrapper.update();

      setTimeout(() => {
        expect(props.onCreate).toHaveBeenCalledTimes(1);
        expect(props.onCreate).toHaveBeenCalledWith({
          claim: 'title',
          slug: 'slug',
          claimant: 1,
          claimant_id: 1,
          rating: 1,
          rating_id: 1,
          claim_date: moment(new Date('2020-12-12')).format('YYYY-MM-DDTHH:mm:ssZ'),
          checked_date: moment(new Date('2020-12-12')).format('YYYY-MM-DDTHH:mm:ssZ'),
          claim_sources: 'claim_sources',
          fact: 'review',
          review_sources: 'review_sources',
          meta: {
            canonical_URL: undefined,
            description: undefined,
            title: undefined,
          },
          description: {
            time: 1595747741807,
            blocks: [
              {
                type: 'header',
                data: {
                  text: 'Editor.js',
                  level: 2,
                },
              },
              {
                type: 'paragraph',
                data: {
                  text:
                    'Hey. Meet the new Editor. On this page you can see it in action — try to edit this text.',
                },
              },
            ],
            version: '2.18.0',
          },
        });
        done();
      }, 0);
    });
    it('should submit form with new name', (done) => {
      const data2 = { ...data, id: 1 };
      data2.meta_fields = {
        sample: 'testing',
      };
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <ClaimCreateForm onCreate={props.onCreate} data={data2} />
          </Provider>,
        );
      });
      act(() => {
        wrapper.find(Collapse).at(2).find('Button').at(0).simulate('click');
      });
      wrapper.update();

      act(() => {
        const input = wrapper.find('FormItem').at(1).find('TextArea').at(0);
        input.simulate('change', {
          target: {
            value:
              'new name new claim title for testing claim update and testing slug update. slug maker should take only first one hundred and fifty characters to create a slug and not all characters from title, since claim title can be  more than 150',
          },
        });
        wrapper.find(Collapse).at(0).find('Button').at(0).simulate('click');

        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Update');
        submitButtom.simulate('submit');
      });

      setTimeout(() => {
        expect(props.onCreate).toHaveBeenCalledTimes(1);
        expect(props.onCreate).toHaveBeenCalledWith({
          claim:
            'new name new claim title for testing claim update and testing slug update. slug maker should take only first one hundred and fifty characters to create a slug and not all characters from title, since claim title can be  more than 150',
          slug:
            'new-name-new-claim-title-for-testing-claim-update-and-testing-slug-update-slug-maker-should-take-only-first-one-hundred-and-fifty-characters-to-creat',
          claimant: 1,
          claimant_id: 1,
          rating: 1,
          rating_id: 1,
          claim_date: moment(new Date('2020-12-12')).format('YYYY-MM-DDTHH:mm:ssZ'),
          checked_date: moment(new Date('2020-12-12')).format('YYYY-MM-DDTHH:mm:ssZ'),
          claim_sources: 'claim_sources',
          fact: 'review',
          review_sources: 'review_sources',
          description: {
            time: 1595747741807,
            blocks: [
              {
                type: 'header',
                data: {
                  text: 'Editor.js',
                  level: 2,
                },
              },
              {
                type: 'paragraph',
                data: {
                  text:
                    'Hey. Meet the new Editor. On this page you can see it in action — try to edit this text.',
                },
              },
            ],
            version: '2.18.0',
          },
          meta_fields: {
            sample: 'testing',
          },
        });
        done();
      }, 0);
    });
    it('should submit form with updated data', (done) => {
      const newDescription = {
        time: 1595747741807,
        blocks: [
          {
            type: 'header',
            data: {
              text: 'Editor.js',
              level: 2,
            },
          },
          {
            type: 'paragraph',
            data: {
              text: 'updated description',
            },
          },
        ],
        version: '2.18.0',
      };
      wrapper.update();
      act(() => {
        wrapper
          .find('FormItem')
          .at(9)
          .find('Editor')
          .at(0)
          .props()
          .onChange({ target: { value: newDescription } });
        wrapper
          .find('FormItem')
          .at(1)
          .find('TextArea')
          .at(0)
          .simulate('change', { target: { value: 'new name' } });
        wrapper
          .find(SlugInput)
          .find('FormItem')
          .find('input')
          .simulate('change', { target: { value: 'new-slug' } });
        wrapper
          .find('FormItem')
          .at(4)
          .find('Select')
          .at(0)
          .props()
          .onChange({ target: { value: 2 } });
        wrapper
          .find('FormItem')
          .at(5)
          .find('Select')
          .at(0)
          .props()
          .onChange({ target: { value: 3 } });
        wrapper
          .find('FormItem')
          .at(7)
          .find(DatePicker)
          .at(0)
          .props()
          .onChange({ target: { value: moment(new Date('2020-01-01')) } });
        wrapper
          .find('FormItem')
          .at(8)
          .find(DatePicker)
          .at(0)
          .props()
          .onChange({ target: { value: moment(new Date('2020-04-04')) } });
        wrapper
          .find('FormItem')
          .at(3)
          .find('TextArea')
          .at(0)
          .simulate('change', { target: { value: 'new review' } });
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Submit');
        submitButtom.simulate('submit');
      });

      setTimeout(() => {
        expect(props.onCreate).toHaveBeenCalledTimes(1);
        expect(props.onCreate).toHaveBeenCalledWith({
          claim: 'new name',
          slug: 'new-slug',
          claimant: 2,
          claimant_id: 2,
          rating: 3,
          rating_id: 3,
          claim_date: moment(new Date('2020-01-01')).format('YYYY-MM-DDTHH:mm:ssZ'),
          checked_date: moment(new Date('2020-04-04')).format('YYYY-MM-DDTHH:mm:ssZ'),
          claim_sources: 'claim_sources',
          fact: 'new review',
          review_sources: 'review_sources',
          description: newDescription,
        });
        done();
      }, 0);
    });

    it('should highlight form field with error in first panel', () => {
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      act(() => {
        wrapper
          .find('FormItem')
          .at(1)
          .find('TextArea')
          .at(0)
          .simulate('change', { target: { value: '' } });
        wrapper
          .find('FormItem')
          .at(4)
          .find('Select')
          .at(0)
          .props()
          .onChange({ target: { value: 2 } });
        wrapper
          .find('FormItem')
          .at(5)
          .find('Select')
          .at(0)
          .props()
          .onChange({ target: { value: 3 } });
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Submit');
        submitButtom.simulate('submit');
      });
      expect(props.onCreate).not.toHaveBeenCalled();
    });
    it('should highlight form field with error in second panel', () => {
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      act(() => {
        wrapper
          .find('FormItem')
          .at(1)
          .find('TextArea')
          .at(0)
          .simulate('change', { target: { value: 'claim' } });
        wrapper
          .find(SlugInput)
          .find('FormItem')
          .find('input')
          .simulate('change', { target: { value: 'new-slug' } });
        wrapper
          .find('FormItem')
          .at(4)
          .find('Select')
          .at(0)
          .props()
          .onChange({ target: { value: 2 } });
        wrapper
          .find('FormItem')
          .at(5)
          .find('Select')
          .at(0)
          .props()
          .onChange({ target: { value: 3 } });
      });
      act(() => {
        const formListInputCount = wrapper.find('FormList').find('Input').length;
        expect(formListInputCount).toBe(0);
        act(() => {
          const button = wrapper.find('FormItem').at(10).find('Button').at(0);
          expect(button.text()).toBe('Add Claim sources');
          button.simulate('click');
        });
      });
      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Submit');
        submitButtom.simulate('submit');
      });
      expect(props.onCreate).not.toHaveBeenCalled();
    });
    it('should add review sources input field on button click', () => {
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      const formListInputCount = wrapper.find('FormList').find('Input').length;
      expect(formListInputCount).toBe(0);
      act(() => {
        const button = wrapper.find('FormItem').at(12).find('Button').at(0);
        expect(button.text()).toBe('Add Review sources');
        button.simulate('click');
      });
      wrapper.update();
      expect(wrapper.find('FormList').find(Input).length).not.toBe(0);
    });
    it('should remove review sources input field on button click', () => {
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      act(() => {
        const button = wrapper.find('FormItem').at(12).find('Button');
        expect(button.text()).toBe('Add Review sources');
        button.simulate('click');
      });
      wrapper.update();
      const formInputFieldCount = wrapper.find('FormList').find(Input).length;
      act(() => {
        wrapper.find('FormList').find(MinusCircleOutlined).at(0).simulate('click');
      });
      wrapper.update();
      expect(wrapper.find('FormList').find(Input).length).toBe(formInputFieldCount - 2);
    });
    it('should add claim sources input field on button click', () => {
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      const formListInputCount = wrapper.find('FormList').find('Input').length;
      expect(formListInputCount).toBe(0);
      act(() => {
        const button = wrapper.find('FormItem').at(11).find('Button').at(0);
        expect(button.text()).toBe('Add Claim sources');
        button.simulate('click');
      });
      wrapper.update();
      expect(wrapper.find('FormList').find(Input).length).not.toBe(0);
    });
    it('should remove claim sources input field on button click', () => {
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      act(() => {
        const button = wrapper.find('FormItem').at(11).find('Button');
        expect(button.text()).toBe('Add Claim sources');
        button.simulate('click');
      });
      wrapper.update();
      const formInputFieldCount = wrapper.find('FormList').find(Input).length;
      act(() => {
        wrapper.find('FormList').find(MinusCircleOutlined).at(0).simulate('click');
      });
      wrapper.update();
      expect(wrapper.find('FormList').find(Input).length).toBe(formInputFieldCount - 2);
    });
    it('should submit with no dates', (done) => {
      const data2 = {
        claim: 'title',
        slug: 'slug',
        claimant: 1,
        rating: 1,
        claim_sources: 'claim_sources',
        fact: 'review',
        review_sources: 'review_sources',
        description: {
          time: 1595747741807,
          blocks: [
            {
              type: 'header',
              data: {
                text: 'Editor.js',
                level: 2,
              },
            },
            {
              type: 'paragraph',
              data: {
                text:
                  'Hey. Meet the new Editor. On this page you can see it in action — try to edit this text.',
              },
            },
          ],
          version: '2.18.0',
        },
      };
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <ClaimCreateForm data={data2} onCreate={props.onCreate} />
          </Provider>,
        );
      });
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Submit');
        submitButtom.simulate('submit');
      });

      setTimeout(() => {
        expect(props.onCreate).toHaveBeenCalledTimes(1);
        expect(props.onCreate).toHaveBeenCalledWith({
          claim: 'title',
          slug: 'slug',
          claimant: 1,
          claimant_id: 1,
          rating: 1,
          rating_id: 1,
          claim_date: null,
          checked_date: null,
          claim_sources: 'claim_sources',
          fact: 'review',
          review_sources: 'review_sources',
          description: {
            time: 1595747741807,
            blocks: [
              {
                type: 'header',
                data: {
                  text: 'Editor.js',
                  level: 2,
                },
              },
              {
                type: 'paragraph',
                data: {
                  text:
                    'Hey. Meet the new Editor. On this page you can see it in action — try to edit this text.',
                },
              },
            ],
            version: '2.18.0',
          },
          meta: {
            canonical_URL: undefined,
            description: undefined,
            title: undefined,
          },
        });
        done();
      }, 0);
    });

    it('should not submit with empty url for review sources', (done) => {
      const data2 = {
        id: 1,
        claim: 'title',
        slug: 'slug',
        claimant: 1,
        rating: 1,
        claim_sources: 'claim_sources',
        fact: 'review',
        description: {
          time: 1595747741807,
          blocks: [
            {
              type: 'header',
              data: {
                text: 'Editor.js',
                level: 2,
              },
            },
            {
              type: 'paragraph',
              data: {
                text:
                  'Hey. Meet the new Editor. On this page you can see it in action — try to edit this text.',
              },
            },
          ],
          version: '2.18.0',
        },
      };
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <ClaimCreateForm data={data2} onCreate={props.onCreate} />
          </Provider>,
        );
      });
      act(() => {
        wrapper.find(Collapse).at(1).find('Button').at(0).simulate('click');
      });
      wrapper.update();
      const formListInputCount = wrapper.find('FormList').find('Input').length;
      expect(formListInputCount).toBe(0);
      act(() => {
        const button = wrapper.find('FormItem').at(12).find('Button').at(0);
        expect(button.text()).toBe('Add Review sources');
        button.simulate('click');
      });
      wrapper.update();
      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        submitButtom.simulate('submit');
        wrapper.update();
      });

      setTimeout(() => {
        expect(props.onCreate).toHaveBeenCalledTimes(0);
        done();
      }, 0);
    });
  });
});
