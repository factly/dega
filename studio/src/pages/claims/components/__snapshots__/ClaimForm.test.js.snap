// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claims Create Form component snapshot testing should match component with data 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ClaimForm
    data={
      Object {
        "checked_date": "2020-12-12T00:00:00.000Z",
        "claim": "title",
        "claim_date": "2020-12-12T00:00:00.000Z",
        "claim_sources": "claim_sources",
        "claimant": 1,
        "description": Object {
          "blocks": Array [
            Object {
              "data": Object {
                "level": 2,
                "text": "Editor.js",
              },
              "type": "header",
            },
            Object {
              "data": Object {
                "text": "Hey. Meet the new Editor. On this page you can see it in action â€” try to edit this text.",
              },
              "type": "paragraph",
            },
          ],
          "time": 1595747741807,
          "version": "2.18.0",
        },
        "fact": "review",
        "rating": 1,
        "review_sources": "review_sources",
        "review_tag_line": "review_tag_line",
        "slug": "slug",
      }
    }
    onCreate={[MockFunction]}
  />
</ContextProvider>
`;

exports[`Claims Create Form component snapshot testing should render the component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ClaimForm />
</ContextProvider>
`;
